domain/externalcontroller/state/state.go

```diff
@@ -27,7 +28,81 @@ func NewState(factory domain.DBFactory) *State {
 	}
 }
 
-func (st *State) UpdateExternalController(ctx context.Context, ci crossmodel.ControllerInfo, modelUUIDs []string) error {
+func (st *State) Controller(
+	ctx context.Context,
+	controllerUUID string,
+) (*crossmodel.ControllerInfo, error) {
+
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var (
+		controllerInfo crossmodel.ControllerInfo
+		controllerRow  *sql.Rows
+	)
+	err = db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
+		q := `
+SELECT alias, ca_cert,  address  
```

Comment:
Let's format these nicely.
```sql
SELECT alias, ca_cert, address  
FROM   external_controller AS ctrl
       LEFT JOIN external_controller_address AS addrs
       ON ctrl.uuid = addrs.controller_uuid
WHERE  ctrl.uuid = ?
```
No need to order the result.

Reply:
Actually, about the order: if I remove it, then  the test `TestRetrieveExternalController` will fail because the two address are in opposite order. 
Maybe the problem is elsewhere and the `ORDER BY` is not a correct fix, but the returned order matters, right?

Reply:
And to be more specific, the returned `address` from `external_controller_address` are ordered by its value (so if I change the second IP in the test to `200` instead of `10.0.0.1` the order will change because `200` is > than `192.168.1.1.`. I guessed this happens because on the `SELECT alias, ca_cert, address` I don't retrieve the address' UUID.